### DISCUSSION TITLE
2022 Community Survey Results

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#9026] [General]
487! That's how many people completed our 2022 community survey; far more than any year previously! A huge thank you to everyone who took the time to share their details and feedback.

Let's dive right into the results.

### 1. Which best describes your organization's industry?

![q1](https://user-images.githubusercontent.com/13487278/161555749-92c879f3-326e-456a-9333-9880bea24df5.png)

We attempted to provide a more granular selection of industries this year, but as with last year we see a heavy skew toward the ISP/telecommunications sector at 27.5%. Education comes in second at 8.2%, followed by cloud services at 7.6%. (Our catch-all category of "other" accounted for only 7.8% of responses, improved from 20% last year.)

### 2. In what country is your organization based?

![q2](https://user-images.githubusercontent.com/13487278/161555775-3dc0e2cc-eac3-4395-be32-e16c4e65ba0f.png)

Survey respondents this year represent 58 unique countries, with the largest NetBox presence by far seen in the United States (21.2%) and Germany (18%). The top ten are listed above.

![q2b](https://user-images.githubusercontent.com/13487278/161555806-a3f4bcdb-923e-4c0c-8039-b5bbe7e365c0.png)

Overall, 3 in 5 respondents are based in Europe.

### 3. How large is your organization (head count)?

![q3](https://user-images.githubusercontent.com/13487278/161555823-43558e97-8c19-4045-90c0-2317d8c81101.png)

### 4. How many NetBox users does your organization have?

![q4](https://user-images.githubusercontent.com/13487278/161555839-e32b3605-31a8-4306-8306-822380c6ec09.png)

### 5. Roughly how long have you been using NetBox?

![q5](https://user-images.githubusercontent.com/13487278/161555852-00b43173-78ce-4a17-bf2d-b68c4bc79754.png)

Not much change in the organizational data as compared to last year.

### 6. What version of NetBox are you running in production?

![q6](https://user-images.githubusercontent.com/13487278/161555869-df67d26e-6d8d-4bd9-b3ea-134985244a41.png)

This was a new question for 2022, and I'm thrilled to see that over 81% of respondents are running NetBox v3.0 or later! This is especially impressive when you consider that the number of new adopters (question five above) account for only 23% of the responses.

The upcoming v3.2 release will offer tons of additional incentive to upgrade for those who have been putting it off.

### 7. How many devices do you manage in NetBox?

![q7](https://user-images.githubusercontent.com/13487278/161555888-826aed6e-4da0-46fa-8e6a-0c1154f88b03.png)

### 8. How many prefixes do you manage NetBox?

![q8](https://user-images.githubusercontent.com/13487278/161555902-8c3f5890-8b25-4816-a8c7-e04cf65f508e.png)

### 9. How have you deployed NetBox?

![q9](https://user-images.githubusercontent.com/13487278/161611641-249ea3ed-d65a-4bd6-81db-6c055c397834.png)

Not much change from last year, but containers seems to comprise a smaller chunk of the install base this time around: 29.4% (cloud- and self-hosted combined) versus 38.2% previously. (Note that the total number of responses is greater than the number of respondents, because some users have deployed NetBox using multiple methods.)

### 10. How do you upgrade NetBox to a new release?

![q10](https://user-images.githubusercontent.com/13487278/161555934-90bf289f-67e4-4e5f-af1c-b9784e8cec6f.png)

Nearly identical to last year's results, with a smaller number of respondents maintaining local forks (2.5% vs 3.7% last year). This could be attributable to increased use of the plugins framework, increased adoption by new users, or both.

### 11. How frequently do you upgrade to a new release? (Best match)

![q11](https://user-images.githubusercontent.com/13487278/161555951-de799a26-3715-4947-bd39-dc5d26916c4d.png)

This year we wanted to ask about upgrade frequency in addition to method. It's very encouraging to see that nearly a third of respondents are keeping up with the latest patch releases. Feedback and bug reports from these early adopters really helps drive development.

### 12. What types of resources do you manage with NetBox?

![q12](https://user-images.githubusercontent.com/13487278/161555965-6b82a728-ef49-4e4a-b308-dd81c06a6561.png)

### 13. Which of the following NetBox features do you use?

![q13](https://user-images.githubusercontent.com/13487278/161555974-8b009983-cd38-441a-8e4d-98e610cfd08d.png)

### 14. What plugins (if any) do you use?

![q14](https://user-images.githubusercontent.com/13487278/161555982-82fa408b-c320-4447-a55d-434f5b9717e4.png)

Here are the top ten plugins selected by respondents. (We're giving absolute counts here rather than percentages as not all respondents use plugins.) Additionally, 55 respondents indicated they they use custom plugins.

## Open-Ended Responses

The last question on our survey was:

> If you could add one feature to NetBox, what would it be?

Out of 487 total respondents, we received 270 answers to this question. Given the nature of free-form responses, these are challenging to quantify, but I did my best to categorize them by general topic.

First, I omitted any response that was already captured in a feature request, implemented in the upcoming v3.2 release, or simply too ambiguous to confidently categorize. Additionally, I recorded only the first item in responses which included multiple suggestions. Listed below are all the topics that ultimately were mentioned at least five times.

* VXLAN/EVPN overlay modeling (17)
* Cable modeling/tracing improvements (15)
* UI/workflow improvements (10)
* Virtual circuit modeling (10)
* Topology diagram generation (9)
* Datacenter floor plan diagrams (7)
* VPN modeling (6)
* DNS zone/record management (5)
* GPON/FTTH modeling (5)
* Geographical map display (5)
* Auto discovery (5)

Right away we recognize some recurring themes from [last year's survey](https://github.com/netbox-community/netbox/discussions/6086): VXLAN/EVPN overlay, VPN, and virtual circuit modeling remain some of the hottest items, along with support for datacenter floor plans and topology diagrams. While these have been on the radar for years, we have yet to dedicate time to devising solid implementation proposals.

It's curious to see such a strong showing for topology diagrams and DNS record management given that there are [community plugins](https://github.com/netbox-community/netbox/wiki/Community-Contributions#plugins) already available for both. We may eventually consider introducing these functions natively within NetBox, but in the near term at least it makes senses to let the plugin developers continue to focus on these features.

UI/workflow improvements is a popular but very broad set of suggestions; we'll need to dig more into these individually and explore what's feasible.

One of the most popular items recently is extending the cabling model to better support simplex-to-duplex fiber connections, which is something I hope to take a closer look at this year.

Interestingly, firewall policy/ACL management---previously a top item---is less in demand this year, and I'm not sure why. It could be that those who need it most have implemented bespoke solutions in the form of custom plugins. Also of note are some topics that have disappeared from the list, namely GraphQL, ASN modeling, and SSO support: These have all been implemented in recent releases.

Thanks again to everyone who completed the survey! This data will be incredibly helpful as we plan out the remainder of our 2022 roadmap.

**Comment by joaolucasmacedo** [comment]
Very good!
I miss the questions regarding how people used to document their environment before Netbox, although it is very clear that most people use spreadsheets.

**Reply by jeremystretch** [reply]
We decided to drop that one as we kept seeing roughly the same results each year, and it doesn't have much impact on our future development at this point.

**Comment by dreng** [comment]
Thank you for sharing the results which are very interesting.

Although I didn't suggest Topology Maps, I'd like to see it hardwired into NetBox. The reasons are quality and integration. None of the plugins are very well thought out and the nextbox-ui-plugin, which is ranked #1 and has a good approach, is not really well maintained. Also heavy bugs sometimes take a long time to get fixed. Sometimes the plugin works partially or not at all after an upgrade from NetBox. It might be a bit much to ask one person to put all his spare time and priority into a plugin, but that seems to be the dilemma here. Anyway, I would expect that you and your team could do a better job here. Especially the point that the feature would always work even after upgrading NetBox is worth it.

**Reply by DanSheps** [reply]
Topology maps were originally taken out of NetBox for a variety of reasons, one of them is simply it is difficult to maintain the map dependencies and keep the code in sync on our end.

I personally can go either way, either keep it a plugin or bring it into core.  However, there is currently a large backlog of FR's and we obviously can't get to them all.

As far as the maintainability of the plugins that do this already, that really is up to the maintainers but perhaps you could suggest they find additional volunteers or you could even volunteer yourself if you have the expertise.

**Comment by jcralbino** [comment]
Hello Jeremy 

Indeed the survey shows that more people are looking ways in how to document vxlan/evpn environmental. 
I would like to see how many people are deploying netbox in a distributed way , where redis service and postgresql are deployed in a different server than netbox. And details on how many of them have a redundant environment with load balancer or even redundant db/redis services

Cheers

**Comment by abdalawood** [comment]
Hello Jeremy,

I would like to know if NetBox has any plans regarding the importation of floor plan images. We really need this functionality because NetBox has not yet been approved for implementation in the company precisely due to the difficulty of inserting floor plan documentation into the software. #5844

**Reply by abhi1693** [reply]
You can install https://github.com/tbotnz/netbox_floorplan and get netbox approved in your company

**Reply by abdalawood** [reply]
Could you tell me where I can get that approval? Because I have already installed this plugin on the netbox here at the company
